olsconfig.yaml

```diff
@@ -11,12 +11,15 @@ LLMProviders:
     models:
       - name: gpt-4-1106-preview
 OLSConfig:
-  enable_debug_ui: false
+  enable_debug_ui: true
   conversation_cache:
     type: memory
     memory:
       max_entries: 1000
-  logger_config:
-    default_level: INFO
+  logging_config:
+    app_log_level: info
+    httpcore_log_level: warning
+    httpx_log_level: warning
+    urllib3_log_level: warning
```

Comment:
this is ok to add to the config yaml, but we should not be exposing these fields on the OLS CRD, it's too much of an internal implementation detail.

perhaps "app_log_level" and "library_log_level" or something.  (Yes that means we'd use the same loglevel for all the libraries)

Reply:
- Can we have one global log level, too?
- Why [just] these packages? Can we make the list of packages dynamic?
- Can we please expose a knob for log level(s) through the API so it can be changed at runtime? It would be invaluable for tracking down production issues once it's in the field

Reply:
> Can we please expose a knob for log level(s) through the API so it can be changed at runtime? It would be invaluable for tracking down production issues once it's in the field

i don't think that this is needed, certainly not right now anyway.  OLS itself is stateless, so i can't think of a scenario in which we'd need to increase the loglevel and would lose critical debugging state if OLS was restarted to change the loglevel.  Do you have a scenario in mind?

I appreciate the idea and i'm not going to claim it would never be useful, but adding something like that means we need to stop and design the interface for changing the log level (who can change it? is it protected by auth?  is it controlled by the operator or CRD somehow?  does that mean new logic in the operator too?) as well as (ideally) test scenarios for it to ensure it functions as desired.  I just don't see the benefits being worth that cost right now

To my knowledge none of the core OCP payload binaries support changing loglevel dynamically other than maybe some networking level pieces.  I won't claim we've never wished we had it, but it's also never been a significant obstacle to supporting our users.

Reply:
I've mostly been thinking about tracking down connectivity and authentication/authorization issues, such as this thing is not talking to that thing and it's not giving a clear reason why, so the CX folks will likely need to ask the CU to enable the highest level of logging for the networking packages/libraries all the way down to OpenSSL, see what the issue is and then restore the normal level of logging. Changing the log level dynamically would start capturing debug data right when the issue is happening. If a restart is needed, it's a less clean situation where something can change due to the restart and the issue would go away for a while or it would take extra effort to reproduce the situation again. But I agree it's not the first priority.

Reply:
I'll do a single setting `library_log_level` now. Once there is a need to further break it down, its easy to change.